### ISSUE TITLE
原型化 Transformer 与观测封闭搜索的最小接口

### ISSUE THREAD

**Issue by titanxxh** [titanxxh/7wd#42]
## Question

在 ZeusAI 迁移研究、观测封闭搜索选择、Transformer 表征研究和资产审计完成后，用不训练长模型的低成本原型比较最少两个端到端接口：从行动玩家 information state 生成组件/合法动作表示，经批量 policy/value 推理进入搜索，再产出有明确 provenance 的 policy/value targets。原型应让用户实际比较观测边界、chance/afterstate、跨 actor batching、模型发布边界、吞吐估算和 CPU 发布 seam，并据此选择一个核心接口。

**Comment by titanxxh** [comment]
## 可交互原型已就绪，等待人工选择

原型文件：`tools/ai/prototypes/transformer_search_interface.html`

本地打开：

```bash
google-chrome file:///home/xxh/7wd/tools/ai/prototypes/transformer_search_interface.html
```

它是单文件、无服务依赖、无持久化的 throwaway logic prototype；没有启动训练、没有读取 Selection/Validation/Test roots，也没有修改 C++ 搜索、训练器或现有 bundle 合同。

### 实际可比较的三臂

1. `dense`：同一个 Small typed-token trunk 后输出 4512 logits，保留兼容对照；
2. `pure candidate`：只对 sorted legal IDs 做 compositional scoring，作为消融；
3. `hybrid candidate`：compositional score + 小 action-ID residual/bias，作为推荐核心。

当前推荐的公共 seam 是：

```text
ActorViewBatchRequest(
  information_state[18134],
  sorted_legal_ids[K],
  policy_id / descriptor hash
)
  -> legal_logits[K] + actor_value
  -> boundary scatter[4512]
  -> observation-closed search
  -> full root visits + legal-terminal actor-relative value
```

Bundle 内部保留 typed tokens、32-latent Small trunk 和 hybrid head；Dense 与 pure candidate 不进入默认发布路线，只承担公平 control/ablation。TorchScript/libtorch 保留兼容基线，AOTInductor 只有通过 ARM64 C++ parity、cold-start 和完整 search P95 后才可替代。

### 原型覆盖的棘手路径

- P1 私有 conspiracy 选择后的 actor-view 输入；
- 两个 compatible referee worlds 的 hidden-world twin closure；
- P0/P1 不同 event/legal 长度的同 generation batching；
- 64/128/256 root-search ladder，全部合法动作和 proof-excluded zero visit 都保留；
- player action → afterstate → exact multi-outcome chance，chance lane 不调用网络；
- extra turn 后仍为同 actor，value perspective 不翻转；
- 私有信息公开披露；
- Political 合法终局后才生成 value target；
- 完整 policy/value provenance seal；
- hidden leak、dropped legal、leaf-value target contamination、mixed generation 四种 INVALID；
- 合同有效但模拟 strength CI 失败时单独归类 NO-GO。

### 真实浏览器烟测

使用仓库已有 Playwright 驱动 `/usr/bin/google-chrome` 点击了完整 15 步推荐链路、hidden-twin、chance、batch 和 INVALID/NO-GO 场景：

- 推荐链路：`CONTRACT-READY`，2 个同 actor decisions 的 visit sums 分别为 `64/64` 与 `128/128`，两个 value target 均来自 `legal_terminal.actor_relative_return`；
- hidden leak：`INVALID`，token/output twin closure 同时失败；
- mixed generation：`INVALID`，generation 原子性失败；
- dropped legal：`INVALID`，legal/target completeness 失败；
- leaf value 污染：`INVALID`，terminal-only value 失败；
- 模拟 strength 失败：`NO-GO`，没有合同 gate 失败；
- 页面脚本/console errors：`0`。

吞吐规划器是明确标注的可编辑假设，不是 benchmark。当前 GPU 有其他 owner，因此没有加载新模型；旧 dense `2.914 ms` batch-1 与历史 `0.143–0.150 games/s` 只显示为参考边界，不能充当新路线 PASS。

请实际打开原型，至少走一次“推荐全链路”和“Hidden-world twins”。若接受推荐核心 seam，请在本票确认；在人工确认前本 HITL 票保持 open，#43 继续被它阻塞。

**Comment by titanxxh** [comment]
用户已在当前 Wayfinder 会话明确回复“接受推荐”。正式采用 actor-view typed-token + fixed-latent Small trunk + hybrid legal-candidate head 的核心接口；dense 仅作同 trunk control，pure candidate 仅作 ablation；candidate logits 在发布边界 scatter 回 4,512 masked logits。TorchScript 保留兼容基线，AOTInductor 只有通过 ARM64 C++ parity、cold-start 与完整 search P95 后才可替代。原型与浏览器证据见上方评论。
